src/main/java/CinemaRoomRestService/RestService/service/BookingService.java

```diff
@@ -0,0 +1,67 @@
+package CinemaRoomRestService.RestService.service;
+
+import CinemaRoomRestService.RestService.*;
+import CinemaRoomRestService.RestService.api.exception.WrongPasswordException;
+import CinemaRoomRestService.RestService.api.exception.SeatOutOfBoundsException;
+import CinemaRoomRestService.RestService.domain.entity.CinemaRoom;
+import CinemaRoomRestService.RestService.domain.entity.Seat;
+import CinemaRoomRestService.RestService.domain.entity.Statistics;
+import CinemaRoomRestService.RestService.domain.entity.TokenOfSeat;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+@NoArgsConstructor
+public class BookingService {
+
+    @Autowired
+    CinemaRoom cinemaRoom;
+
+    public CinemaRoom getSeatInfo() {
+        return cinemaRoom;
+    }
+
+    private Seat getSeatAfterBooking(TokenOfSeat tokenOfSeat) {
+        Seat seat = tokenOfSeat.getTicket();
+        seat.isOccupied = false;
+        return seat;
+    }
+
+    private TokenOfSeat getSeatAfterReturning(Seat seat) {
+        seat.isOccupied = true;
+        return new TokenOfSeat(seat);
+    }
+
+    public TokenOfSeat bookingOfSeat(Seat seat) {
+        if (seat.getRow() > cinemaRoom.getTotalRows() || seat.getColumn() > cinemaRoom.getTotalColumns()
```

Comment:
to many logic inside that method, delegate that to other private method, hide that if

Reply:
other private method
    private boolean seatExists(Seat seat) {
        return seat.getRow() > cinemaRoom.getTotalRows() || seat.getColumn() > cinemaRoom.getTotalColumns()
                || seat.getRow() < 1 || seat.getColumn() < 1;
    }

Reply:
if (!seatExists(seat)) throw new SeatOutOfBoundsException(ErrorMessage.OUT_OF_BOUNDS.toString());

Reply:
Fixed